### DISCUSSION TITLE
Moving Beets library and files from macOS to Linux (Unraid) server

### DISCUSSION THREAD

**Discussion by j-peeters** [beetbox/beets#5127] [Q&A] [answered]
Hi all, I'm transitioning from a Mac Mini as my media server to a Intel NUC Unraid machine. 
I hope to find some guidance on how to best move my music library and database to the new server.

My library consists of multiple thousands of albums all tagged via `beet import -t` and I would love to not have to import -t them again nor have them autotagged. All tags should be all right. 

`beet move` seemed to be a good way, but I'm uncertain about the following:
I see three options:
1. move the files with the command line from macOS to a remote Unraid share or
2. move the files with the command line to Unraid from a remote macOS share
Alternatively I could 
3. Copy the files manually to a new directory on the Unraid server and run `beet import -AC`.

What I'm not sure about is that:
a) Option 1: If I would do the move from the macOS machine, then the udated database still resides on macOS and I'm not sure how Beets would handle the new paths (of the Unraid disc) when I would move the database manually?
b) Option 2: I first have to move the existing config.yaml file manually to the Unraid server I guess and set the new paths for library and database. Are there any disadvantages in doing this?
c) Option 3: On the Unraid machine the database is still empty so I guess I should use something like `beet import -AC` The `-A` would just import, not autotag. And the `-C` flag would leave the files in the place where I manually copied them. Are there any disadvantages in doing this?

I've considered moving all music files and database manually and thought to change the location in the config.yaml. But I read on a Reddit post that that might not work because the file paths are not relative? is this still true? A search/replace in the database could maybe solve that but I read that was disadvised and rightly so, I'm not yet well versed in Linux paths so I could easily write the paths in the wrong way.

So hopefully any of you has experience with this or knows how to do this with as less as possible effort ;-)
Thanks for any suggestions and have a great day, Jan

**Comment by JOJ0** [comment, marked as answer]
A pragmitacal approach if you are not afraid of / know some SQL

- Copy everything over to your new server! Database and Files

- Adapt what's required in config.yaml. Should only be a couple of paths.

- Create an SQL command that replaces the portion of your path in the beets library (https://www.sqlite.org/lang_update.html)


If copying over ALL files is too much or a space issue, test this process with a couple of albums only. If it worked, mount your existing data storage to the new server or just copy the rest. beets will basically work if your files are not entirely there already. If you try file manipulations of missing files you obviously will get IO errors. So just test if it's working with albums you are sure you have copied for testing already.

All that said: IF you are aware of some SQL and have a thorough backup strategy! A simple strategy I would do is: Just not touch the existing server. Leave it as-is! Experiment with copies only!

**Reply by j-peeters** [reply]
Thanks @JOJ0 for confirming that search/replace can be done (with the right paths of course).  That's very helpful. I'll give it a try. Thanks!
